src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -569,4 +663,67 @@ private void CheckDisposed()
             throw new ObjectDisposedException(this.GetType().FullName);
         }
     }
+
+    private void OnRedisError(Exception exception, IDatabase cache)
+    {
+        if (exception is RedisConnectionException or SocketException)
+        {
+            var utcNow = DateTimeOffset.UtcNow;
+            var previousConnectTime = ReadTimeTicks(ref _lastConnectTicks);
+            TimeSpan elapsedSinceLastReconnect = utcNow - previousConnectTime;
+
+            // We want to limit how often we perform this top-level reconnect, so we check how long it's been since our last attempt.
+            if (elapsedSinceLastReconnect < ReconnectMinInterval)
+            {
+                return;
+            }
+
+            var firstErrorTime = ReadTimeTicks(ref _firstErrorTimeTicks);
+            if (firstErrorTime == DateTimeOffset.MinValue)
+            {
+                WriteTimeTicks(ref _firstErrorTimeTicks, utcNow);
+                WriteTimeTicks(ref _previousErrorTimeTicks, utcNow);
```

Comment:
Potential races here are not a problem. Correct?
Please add a comment for this.